### DISCUSSION TITLE
连续读取数据

### DISCUSSION THREAD

**Discussion by guchainews** [LUJIAN2020/UsbSerialForAndroid.Net#7] [Q&A] [answered]
大神，能否写一个连续读取数据的DEMO。我照着https://github.com/anotherlab/UsbSerialForAndroid写了一个，使用的是 Dispatcher.UIThread   while(true) { step() } ，但是只能读一次数据，第二次就报错: "Read failed,Result:-1,Address:132,Direction:In,EndpointNumber:4,Type:XferBulk,MaxPacketSize:64,Buffer:,Offset:0,Length:2048,Timeout:10000"

**Comment by LUJIAN2020** [comment, marked as answer]
> 大神，能否写一个连续读取数据的DEMO。我照着[https://github.com/anotherlab/UsbSerialForAndroid写了一个，使用的是](https://github.com/anotherlab/UsbSerialForAndroid%E5%86%99%E4%BA%86%E4%B8%80%E4%B8%AA%EF%BC%8C%E4%BD%BF%E7%94%A8%E7%9A%84%E6%98%AF) Dispatcher.UIThread while(true) { step() } ，但是只能读一次数据，第二次就报错: "Read failed,Result:-1,Address:132,Direction:In,EndpointNumber:4,Type:XferBulk,MaxPacketSize:64,Buffer:,Offset:0,Length:2048,Timeout:10000"

如果没有数据可以读取的话是会报错的，这个需要自己处理，下面是读取的代码
![image](https://github.com/user-attachments/assets/94d9090d-40c7-4ca0-99a9-3d25b4b709d4)


Mono.Android的注释是 传输的数据长度(或零)表示成功，或负值表示失败，我这里做了抛出异常处理
![image](https://github.com/user-attachments/assets/20ee5176-a718-4f6d-a6c5-a39bfa6703f8)

**Reply by guchainews** [reply]
没想到居然返回空就可以了。万分感谢。

**Comment by LUJIAN2020** [comment]
现在我更新了一些代码，抛出异常代码已经删除，如果读不到数据将返回空
